src/language/rule/abstractRule.ts

```diff
@@ -67,11 +67,13 @@ export abstract class AbstractRule implements IRule {
     ): RuleFailure[];
     protected applyWithFunction<T, U>(
         sourceFile: ts.SourceFile,
-        walkFn: (ctx: WalkContext<T | void>, programOrChecker?: U) => void,
```

Comment:
🤔 I think the takeaway from this rule's changes is that the type should be `T | undefined`. That would let you get away with not changing `undefined` to `{}` in the many places elsewhere.

Thoughts @timocov? I don't want to keep blocking this otherwise great rule on type system shenanigans, so if you don't have time to change this I can do so and merge the rule in.

Reply:
> I think the takeaway from this rule's changes is that the type should be T | undefined. That would let you get away with not changing undefined to {} in the many places elsewhere.

It's debating point because of semantic of options, but ok, let's make here `T | undefined` instead of `T | {}`.

See 0bff08430763ebd5af3fa94099a418b228536fff